flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -66,13 +66,14 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
     bool setRouteNameAsTransaction = false,
     RouteNameExtractor? routeNameExtractor,
     AdditionalInfoExtractor? additionalInfoProvider,
+    @visibleForTesting SentryNative? sentryNative,
```

Comment:
If this is a singleton, we could use the `reset` method to set/assert and reset things, no?

Reply:
We could also mock the values from `SentryNativeChannel`

Reply:
I mean.

// create navigation observer
// SentryNative().setSentryNativeChannel(mock)
// assert everything you want
// SentryNative().reset()

Reply:
We verify if the correct methods got called, this is not something we can assert on `SentryNative`. Mocking `SentryNativeChannel` and verifying its methods works though. Overall it's really difficult to do testing properly with the internal singleton use.